glDemo/main.cpp

```diff
@@ -111,21 +160,106 @@ int main() {
 
 
 // renderScene - function to render the current scene
+
 void renderScene()
 {
 	// Clear the rendering window
 	glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
 
-	// Render objects here...
-	//drawShadedTriangle();
-	//drawStarOutline();
-	//drawStarShaded();
-	//drawTank();
-	drawSemiCircleStudio();
+	//randomPoints();
+	edgedRectangles();
+	//blendingBoxes();
 }
 
+
+
 // Rendering functions
 
+void blendingBoxes() 
+{
+	glBegin(GL_QUADS);
+
+	glEnd();
+}
+
+void edgedRectangles() 
+{
+	glLineWidth(5.0f);
+	glBegin(GL_LINES);
+
+	
+	glColor3f(1.0f, 0.0f, 0.0f);
+	glVertex2f(-0.5f, -0.5f);
+
+	glColor3f(1.0f, 1.0f, 0.0f);
+	glVertex2f(0.5f, -0.5f);
+
+	glEnd();
+
+	////////////////////////////
+
+	glLineWidth(10.0f);
+	glBegin(GL_LINES);
+
+
+	glColor3f(0.0f, 0.0f, 1.0f);
+	glVertex2f(0.5f, 0.5f);
+
+	glColor3f(0.0f, 1.0f, 1.0f);
+	glVertex2f(-0.5f, 0.5f);
+
+	glEnd();
+
+	////////////////////////////
+
+	glLineWidth(20.0f);
+	glBegin(GL_LINES);
+
+
+	glColor3f(0.0f, 1.0f, 1.0f);
+	glVertex2f(-0.5f, -0.5f);
+
+	glColor3f(1.0f, 0.0f, 1.0f);
+	glVertex2f(-0.5f, 0.5f);
+
+	glEnd();
+
+	////////////////////////////
+
+	glLineWidth(2.0f);
+	glBegin(GL_LINES);
+
+
+	glColor3f(0.0f, 0.0f, 1.0f);
+	glVertex2f(0.5f, -0.5f);
+
+	glColor3f(0.0f, 1.0f, 1.0f);
+	glVertex2f(0.5f, 0.5f);
+
+	glEnd();
+}
+
+void randomPoints() 
```

Comment:
Your random points function should merge the 2 for loops.  At the moment you iterate through all 100 points setting the size and colour.  Then you iterate through all the points and draw them.  However, they'll all be drawn with the size and colour that was set in the last iteration of the previous loop.

The structure should be

for i = 0..100
{
 Set colour
 Set size
 glBegin points
   Draw single point
 glEnd
}